### DISCUSSION TITLE
DHT Queries don't show the peer executing the query

### DISCUSSION THREAD

**Discussion by umgefahren** [libp2p/rust-libp2p#2199] [Q&A]
When I query the DHT from a Peer, i.e. looking for a provider, why doesn't the peer querying shows up when it's the one providing?
I guess this is intentional, but how can I fix this from an application perspective?

**Comment by mxinden** [comment]
After a quick look I don't think we explicitly filter out the local node from the provider set. I might be wrong. 

Could you provide a small reproducer, e.g. based on the test below?

https://github.com/libp2p/rust-libp2p/blob/25004c46971fd6de1d8eac7f4895e673d9afc95e/protocols/kad/src/behaviour/test.rs#L839-L843

**Reply by umgefahren** [reply]
Although this is not my code, the problem can be seen in [examples/distributed-key-value-store.rs].

**Reply by umgefahren** [reply]
Any updates on this?

**Reply by mxinden** [reply]
Thanks. I seem to have misunderstood your initial question.

Yes, a `get_providers` call does not check the local store. A `get_record` does:

https://github.com/libp2p/rust-libp2p/blob/25004c46971fd6de1d8eac7f4895e673d9afc95e/protocols/kad/src/behaviour.rs#L673-L682

Do you want to propose a patch, making `get_providers` check the local store like `get_records` does today @umgefahren?

**Reply by umgefahren** [reply]
I can certainly try; however I'm not really familiar with the library. If I make the patch it will take time, as I've already found a botch for my application.
